storage/preferences/src/main/java/de/danoeh/antennapod/storage/preferences/SleepTimerPreferences.java

```diff
@@ -45,11 +49,26 @@ public static String lastTimerValue() {
         return prefs.getString(PREF_VALUE, DEFAULT_LAST_TIMER);
     }
 
-    public static long timerMillis() {
+    protected static long timerMillis() {
         long value = Long.parseLong(lastTimerValue());
         return TimeUnit.MINUTES.toMillis(value);
     }
 
+    public static long timerMillisOrEpisodes() {
+        return switch (getSleepTimerType()) {
+            case CLOCK -> timerMillis();
+            case EPISODES -> Long.parseLong(SleepTimerPreferences.lastTimerValue());
+        };
+    }
+
+    public static SleepTimerType getSleepTimerType() {
+        return SleepTimerType.valueOf(prefs.getString(PREF_TIMER_TYPE, DEFAULT_TIMER_TYPE).toUpperCase(Locale.US));
```

Comment:
Please add an "id" parameter to the enum values and store the integer instead. See for example `AutoDeleteAction`